src/fsrs/default.ts

```
@@ -1,62 +1,72 @@
-  Object.freeze([0.001, 5.0]) /** fail stability (multiplier) */,
-  Object.freeze([0.001, 0.25]) /** fail stability (negative power) */,
-  Object.freeze([0.001, 0.9]) /** fail stability (power) */,
-  Object.freeze([0.0, 4.0]) /** fail stability (exponent) */,
-  Object.freeze([0.0, 1.0]) /** stability (multiplier for Hard) */,
-  Object.freeze([1.0, 6.0]) /** stability (multiplier for Easy) */,
-  Object.freeze([0.0, 2.0]) /** short-term stability (exponent) */,
-  Object.freeze([0.0, 2.0]) /** short-term stability (exponent) */,
-]) as readonly (readonly [number /** min */, number /** max */])[]
+    w17_w18_ceiling = clamp(+value.toFixed(8), 0.01, 2.0)
+  }
+  const clip = CLAMP_PARAMETERS(w17_w18_ceiling)
+  return clip.map(([min, max], index) => clamp(parameters[index], min, max))
+}
 
-export const generatorParameters = (
-  props?: Partial<FSRSParameters>
-): FSRSParameters => {
-  let w: number[] = [...default_w]
-  if (props?.w) {
-    if (props.w.length === 19) {
-      w = [...props.w]
-    } else if (props.w.length === 17) {
-      w = props?.w.concat([0.0, 0.0])
+export const migrateParameters = (
+  parameters?: number[] | readonly number[]
+) => {
+  if (parameters === undefined) {
+    return [...default_w]
+  }
+  switch (parameters.length) {
+    case 21:
+      return [...parameters]
+    case 19:
+      console.debug('[FSRS-6]auto fill w from 19 to 21 length')
+      return [...parameters, 0.0, FSRS5_DEFAULT_DECAY]
+    case 17: {
+      const w = [...parameters]
       w[4] = +(w[5] * 2.0 + w[4]).toFixed(8)
       w[5] = +(Math.log(w[5] * 3.0 + 1.0) / 3.0).toFixed(8)
       w[6] = +(w[6] + 0.5).toFixed(8)
-      console.debug('[FSRS V5]auto fill w to 19 length')
+      console.debug('[FSRS-6]auto fill w from 17 to 21 length')
+      return w.concat([0.0, 0.0, 0.0, FSRS5_DEFAULT_DECAY])
     }
+    default:
+      console.warn('[FSRS]Invalid parameters length, using default parameters')
```

Should it throw an error here instead of a warning like it does in the rs version?

https://github.com/open-spaced-repetition/fsrs-rs/blob/6d0dc87ce9092cb8034da57102cf2e2da5cf8f5d/src/model.rs#L300

Would silently using the default parameters be worse than forcing the user to correct them?

---

In some frameworks(ex: vue), parameters might be bound for page rendering, and throwing an error could prevent the page from displaying properly.